breathecode/certificate/tests/actions/tests_certificate_set_default_issued_at.py

```diff
@@ -0,0 +1,227 @@
+"""
+Tasks tests
+"""
+from unittest.mock import patch, call, MagicMock
+from ...actions import certificate_set_default_issued_at
+from ..mixins import CertificateTestCase
+from ...models import UserSpecialty
+
+from django.utils import timezone
+
+
+class ActionCertificateSetDefaultIssuedAtTestCase(CertificateTestCase):
+    def test_issued_at_null_status_error(self):
+
+        model = self.generate_models(user_specialty=True,
+                                     user_specialty_kwargs={
+                                         'status': 'ERROR',
+                                         'issued_at': None
+                                     })
+        query = UserSpecialty.objects.filter(status='PERSISTED', issued_at__isnull=True)
+
+        result = certificate_set_default_issued_at()
+
+        self.assertEqual(list(result), list(query))
+        self.assertEqual(
+            self.all_user_specialty_dict(),
+            self.remove_is_clean([{
+                **self.model_to_dict(model, 'user_specialty'),
+                'status': 'ERROR',
+                'issued_at': None,
+            }]))
+
+    def test_issued_at_set_status_error(self):
+
+        now = timezone.now()
+
+        model = self.generate_models(user_specialty=True,
+                                     user_specialty_kwargs={
+                                         'status': 'ERROR',
+                                         'issued_at': now
+                                     })
+
+        query = UserSpecialty.objects.filter(status='PERSISTED', issued_at__isnull=True)
+
+        result = certificate_set_default_issued_at()
+
+        self.assertEqual(list(result), list(query))
+        self.assertEqual(
+            self.all_user_specialty_dict(),
+            self.remove_is_clean([{
+                **self.model_to_dict(model, 'user_specialty'),
+                'status': 'ERROR',
+                'issued_at': now,
+            }]))
+
+    def test_issued_at_null_status_persisted_one_item(self):
+
```

Comment:
Comment that specifies that the issued_at should remain None because the user_specialty has not cohort specified and its impossible to determine the cohort ending_at